src/libcore/num/mod.rs

```diff
@@ -569,7 +569,7 @@ macro_rules! int_impl {
         #[inline]
         pub fn abs(self) -> $T {
             if self.is_negative() {
-                self.wrapping_neg()
+                -self
```

Comment:
Can you leave a comment saying that this has the correct overflow semantics *only if* kept as `inline` given how std is distributed? I hope we can do something more robust in the future.

Otherwise r=me.